argoverse/data_loading/argoverse_stereo_loader.py

```diff
@@ -0,0 +1,459 @@
+from typing import Dict, Iterator, List, Optional, Union
+
+import cv2
+import numpy as np
+
+from argoverse.utils.calibration import Calibration, load_stereo_calib
+from argoverse.utils.camera_stats import RECTIFIED_STEREO_CAMERA_LIST
+
+logger = logging.getLogger(__name__)
+
+STEREO_FRONT_LEFT_RECT = RECTIFIED_STEREO_CAMERA_LIST[0]
+STEREO_FRONT_RIGHT_RECT = RECTIFIED_STEREO_CAMERA_LIST[1]
+
+STEREO_DISPARITY_LIST = ["stereo_front_left_rect_disparity", "stereo_front_left_rect_objects_disparity"]
+
+DISPARITY_NORMALIZATION: float = 2.0 ** 8  # 256.0
+
+
+class ArgoverseStereoLoader:
+    def __init__(self, root_dir: str, split_name: str) -> None:
+        # initialize class member
+        self.RECTIFIED_STEREO_CAMERA_LIST = RECTIFIED_STEREO_CAMERA_LIST
+        self._log_list: Optional[List[str]] = None
+        self._image_list: Optional[Dict[str, Dict[str, List[str]]]] = None
+        self._disparity_list: Optional[Dict[str, Dict[str, List[str]]]] = None
+        self._image_timestamp_list: Optional[Dict[str, Dict[str, List[int]]]] = None
+        self._timestamp_image_dict: Optional[Dict[str, Dict[str, Dict[int, str]]]] = None
+        self._calib: Optional[Dict[str, Dict[str, Calibration]]] = None
+        self.counter: int = 0
+
+        self.image_count: int = 0
+
+        self.root_dir: str = root_dir
+        self.split_name: str = split_name
+
+        self.current_log = self.log_list[self.counter]
+
+        assert self.image_list is not None
+
+        # load calibration file
+        self.calib_filename: str = os.path.join(
+            self.root_dir,
+            "rectified_stereo_images_v1.1",
+            self.split_name,
+            self.current_log,
+            "vehicle_calibration_stereo_info.json",
+        )
+
+        # stereo camera @5hz
+        self.num_stereo_camera_frame: int = len(self.image_timestamp_list[STEREO_FRONT_LEFT_RECT])
+
+        assert self.calib is not None
+
+    @property
+    def calib(self) -> Dict[str, Calibration]:
+        """Get calibration dict for current log
+
+        Returns:
+            calib: Calibration object for the current log
+        """
+        self.__ensure_calib_is_populated()
+        assert self._calib is not None
+        return self._calib[self.current_log]
+
+    def __ensure_calib_is_populated(self) -> None:
+        """Load up calibration object for all logs
+
+        Returns:
+            None
+        """
+        if self._calib is not None:
+            return
+
+        self._calib = {}
+        for log in self.log_list:
+            calib_filename = os.path.join(
+                self.root_dir,
+                "rectified_stereo_images_v1.1",
+                self.split_name,
+                log,
+                "vehicle_calibration_stereo_info.json",
+            )
+            self._calib[log] = load_stereo_calib(calib_filename)
+
+    @property
+    def log_list(self) -> List[str]:
+        """Return list of log (str) in the current dataset directory
```

Comment:
typo: "Return list of log_ids in the current dataset directory"

The type is already self-evident from the function signature, I believe